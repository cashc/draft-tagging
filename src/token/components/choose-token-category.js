import { h } from 'react-hyperscript-helpers';
import { connect } from 'react-redux';

import { setSelectedTokenCategory } from '../action-creators';
import { DELETE_TOKEN_CATEGORY } from '../constants';
import { getCategoryItemsSorted } from '../util';

function buildCategoryActions(onClick) {
  const categoryItems = getCategoryItemsSorted();
  const categoryActions = categoryItems.map((categoryItem) =>
    createCategoryItemAction(categoryItem, onClick),
  );

  const remover = {
    type: 'CONTEXT_MENU_ITEM_TYPE',
    label: 'Remove',
    name: DELETE_TOKEN_CATEGORY,
    onClick: () => onClick(DELETE_TOKEN_CATEGORY),
  };
  return [...categoryActions, remover];
}

function createCategoryItemAction(categoryItem, onClick) {
  const itemType = 'CONTEXT_MENU_ITEM_TYPE';
  const { categoryType, categoryTitle } = categoryItem;
  return {
    type: itemType,
    label: categoryTitle,
    name: categoryType,
    onClick: () => onClick(categoryType),
  };
}

export function ChooseTokenCategory({
  // clickPosition,
  // close = () => {},
  handleChooseTokenCategory,
}) {
  const actions = buildCategoryActions(handleChooseTokenCategory);

  return h('div', actions.map((a) => h('div', a.categoryTitle)));
}

function mapDispatchToProps(dispatch, { selectedToken, close }) {
  return {
    handleChooseTokenCategory: (category) => {
      dispatch(setSelectedTokenCategory({ selectedToken, category }));
      close();
    },
  };
}

export const ChooseTokenCategoryConn = connect(null, mapDispatchToProps)(
  ChooseTokenCategory,
);
