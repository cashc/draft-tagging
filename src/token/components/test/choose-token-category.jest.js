import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { h } from 'react-hyperscript-helpers';

import { ChooseTokenCategory } from '../choose-token-category';

describe('ChooseTokenCategory', () => {
  describe('when rendering', () => {
    const wrapper = shallow(h(ChooseTokenCategory));
    const json = toJson(wrapper);

    it('should render the menu', () => {
      expect(json).toMatchSnapshot();
    });
  });
});
