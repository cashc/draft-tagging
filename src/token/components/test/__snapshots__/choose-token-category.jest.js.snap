// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseTokenCategory when rendering should render the menu 1`] = `
<div>
  <ContextMenu
    actions={
      Array [
        Object {
          "label": "Name",
          "name": "user_name",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "label": "Job Title",
          "name": "job_title",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "label": "Company Name",
          "name": "company",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "label": "City",
          "name": "city",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "label": "State / Province",
          "name": "state",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "label": "Domain",
          "name": "domain",
          "onClick": [Function],
          "type": "item",
        },
        Object {
          "type": "divider",
        },
        Object {
          "label": "Remove",
          "name": "delete_token",
          "onClick": [Function],
          "type": "item",
        },
      ]
    }
    animate={false}
    clickPosition={Object {}}
    handleClose={[Function]}
    open={true}
  />
</div>
`;
